File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/tasklistview/buildlogic/TaskListViewInjector.java
package com.wiseassblog.samsaradayplanner.ui.tasklistview.buildlogic;

import com.wiseassblog.samsaradayplanner.StorageServiceLocator;
import com.wiseassblog.samsaradayplanner.ui.tasklistview.ITaskListViewContract;
import com.wiseassblog.samsaradayplanner.ui.tasklistview.TaskListView;
import com.wiseassblog.samsaradayplanner.ui.tasklistview.TaskListViewLogic;
import com.wiseassblog.samsaradayplanner.ui.tasklistview.TaskListViewModel;

public class TaskListViewInjector {
    public static void build(
            TaskListView view,
            StorageServiceLocator locator
    ) {

        ITaskListViewContract.ViewModel vm = new TaskListViewModel();

        view.setLogic(
                new TaskListViewLogic(
                        view,
                        vm,
                        locator.getTaskStorageImpl()
                )
        );
    }
}

File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/tasklistview/buildlogic/TaskListViewInjector.kt
package com.wiseassblog.samsaradayplanner.ui.tasklistview.buildlogic

import com.wiseassblog.samsaradayplanner.StorageServiceLocator
import com.wiseassblog.samsaradayplanner.ui.tasklistview.*

internal fun TaskListActivity.buildComponents(
    view: TaskListView,
    locator: StorageServiceLocator
) {
    val vm: ITaskListViewContract.ViewModel = TaskListViewModel()
    view.setLogic(
        TaskListViewLogic(
            view,
            vm,
            locator.taskStorageImpl
        )
    )
}